casper/src/main/scala/coop/rchain/casper/blocks/proposer/BlockCreator.scala

```diff
@@ -150,7 +151,9 @@ object BlockCreator {
                   shardId,
                   casperVersion
                 )
-                _           <- Span[F].mark("block-created")
+                _ <- Span[F].mark("block-created")
+                // signedBlock add signature and replace hashed-without-signature
+                // blockHash to hashed-with-signature blockHash
```

Comment:
This whole confusion seems to be because _sender_ is not added for genesis block, which doesn't make sense why? Genesis master created the genesis block and can be added as a sender and sign. Other genesis validators will validate this block in the same way as any other block.

Also I couldn't find the reason why signing algorithm is included for hashing. In verification is only used here, selected from hard coded algorithms.
https://github.com/rchain/rchain/blob/a37ff7ef5172a370621ab415a33c39a33126030a/casper/src/main/scala/coop/rchain/casper/Validate.scala#L81-L85
Another questions are. Why signing algorithm is included in the block and why signing algorithm is stored and not hashing algorithm how is used with enveloped/detached signatures.
https://en.wikipedia.org/wiki/XML_Signature
https://www.researchgate.net/figure/4-Enveloped-enveloping-and-detached-signatures_fig12_33760566

Every block is hashed without a signature. Signature means signing of the block hash.
`BlockCreator#packageBlock` can accept sender and create block hash (the one and only) and signing should just add _signature_.

But this seems to much for now. I'll create an issue for HF2.